### ISSUE TITLE
`LocalDateTime.of()` throws since month and day 0 are not valid

### ISSUE THREAD

**Issue by InExtremaRes** [js-joda/js-joda#354]
This was mentioned in #349 but the true issue is not with TS declarations but with actual implementation.

When calling `LocalDateTime.of()` without arguments it fails with:

```plain
DateTimeException: Invalid value for MonthOfYear (valid values 1 - 12): 0
```

The issue is that `LocalDateTime.of()`, that internally calls `LocalDateTime.ofNumbers(...)`, assigns a default argument of 0 to every parameter and there is no month nor day 0.

I would expect a `NullPointerException` instead since I think there should not be a default date. A default time of `00:00:00` is quite reasonable, though. So `LocalDateTime.of(...)` should take at least three arguments for the date, leaving de time part optional. The TS declarations should be updated as well.

**Comment by pithu** [comment]
Looks like a bug. threeten takes either two args (date and time) or at least 5 number values, i agree that its ok to expect at least 3 number values year, month and day and set time to start of day in that case. So actually it should probably be implemented like ZoneDateTime.of.

**Comment by InExtremaRes** [comment]
OK, I'll send a PR with the fix and in the meanwhile I'll adjust the TS declarations.

**Comment by pithu** [comment]
Fixed by #357 and #359
